docs/android/codestyle.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "Android Code Style"
+sidebar_label: "Code Style"
+---
+
+## Why Enforcing Code Style
+
+We aim to maintain a **consistent and standard codebase**. By enforcing code style:
+
+- We reduce unnecessary comments on PRs, allowing reviewers to focus on the logic rather than formatting.
+- We ensure the code is easier to read and maintain.
+
+## Language Guidelines
+
+- All code must be written in **English**.
+- Avoid typos and grammatical mistakes. While mistakes are acceptable (as many contributors are non-native speakers), reviewers are encouraged to suggest corrections.
+- Use tools like LLMs or spell checkers to assist in correcting mistakes.
+
+## KTLint Formatter
+
+We use [KTLint](https://pinterest.github.io/ktlint) to enforce Kotlin code style. It is integrated into our Gradle modules and configured via the `.editorconfig` file.
+
+### Custom Rules
+
+- We override some KTLint rules when necessary. To enable or disable a rule:
+  1. Submit a **dedicated PR** with a proper explanation of the change.
+  2. If the change affects a large portion of the codebase, create **two commits**:
```

Comment:
Why not always use two commits?

Reply:
I'll rephrase it was not clear that if there are no changes there is only one commit my bad.

Reply:
My suggestion was more like: even if the rule change only changes a couple of lines of code, use two commits so the rule change is always commit 1 and the changed code is always commit 2.

Reply:
I think my changes fits your suggestion :)